@xen-orchestra/vmware-explorer/esxi.mjs

```diff
@@ -51,7 +54,8 @@ export default class Esxi extends EventEmitter {
 
   async download(dataStore, path, range) {
     strictEqual(this.#ready, true)
-    const url = `https://${this.#host}/folder/${path}?dsName=${dataStore}`
+    notStrictEqual(this.#dcPath, undefined)
+    const url = `https://${this.#host}/folder/${path}?dcPath=${this.#dcPath}&dsName=${dataStore}`
```

Comment:
Are we sure this all values (`#host`, `path`, `#dcPath`, `dataStore`) are safe to use in an URL?

Either use `encodeURIComponent` or even better, use `URL` to build your URL:

```js
const url = new URL('https://localhost')
u.host = this.#host
u.pathname = '/foder/' + path
u.searchParams.set('dc', this.#dcPath)
u.searchParams.set('dsName', dataStore)
```